middleware/decompress.go

```diff
@@ -86,35 +86,51 @@ func DecompressWithConfig(config DecompressConfig) echo.MiddlewareFunc {
 			if config.Skipper(c) {
 				return next(c)
 			}
-			switch c.Request().Header.Get(echo.HeaderContentEncoding) {
-			case GZIPEncoding:
-				b := c.Request().Body
-
-				i := pool.Get()
-				gr, ok := i.(*gzip.Reader)
-				if !ok {
-					return echo.NewHTTPError(http.StatusInternalServerError, i.(error).Error())
-				}
 
-				if err := gr.Reset(b); err != nil {
-					pool.Put(gr)
-					if err == io.EOF { //ignore if body is empty
-						return next(c)
-					}
-					return err
-				}
-				var buf bytes.Buffer
-				io.Copy(&buf, gr)
+			if c.Request().Header.Get(echo.HeaderContentEncoding) != GZIPEncoding {
+				return next(c)
+			}
 
-				gr.Close()
-				pool.Put(gr)
+			b := c.Request().Body
+
+			i := pool.Get()
+			gr, ok := i.(*gzip.Reader)
+			if !ok {
+				return echo.NewHTTPError(http.StatusInternalServerError, i.(error).Error())
```

Comment:
Since you're trying reading the request body, wouldn't this be a 4xx?

Reply:
This was as is. Regardless I believe this is correct.
Since sync.Pool returns an `interface{}` we have no choice to cast the value to the underlying type to work with it.
This middleware creates a sync.Pool to amortise allocation of gzip.Reader objects. If for some reason our pool is returning something else that we aren't expecting than something is wrong with the lib, not the request. Hence the 500. 

An argument could be made that this might even be a good spot to `panic`, but I don't want to introduce unneeded changes in this PR.